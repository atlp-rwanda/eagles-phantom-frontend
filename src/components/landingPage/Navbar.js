import React,{ Component } from "react";
import '../../scss/landingNav.scss';
import { Button, Navbar, Nav, NavItem, NavDropdown, MenuItem,Form } from 'react-bootstrap';

export default class Navbars extends Component{
render(){
  return (
    <Navbar className="nav" collapseOnSelect expand="lg"  variant="dark">
        <Navbar.Brand href="#home">Phantom</Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
   <Navbar.Collapse id="responsive-navbar-nav">
    <Nav className="justify-content-end">
      <Nav.Item className="nav_link" >
      <Nav.Link  href="#">Home</Nav.Link>
      </Nav.Item>
      <Nav.Item>
      <Nav.Link href="AboutUs">About</Nav.Link>
      </Nav.Item>
      <Nav.Item>
      <Nav.Link href="/ContactUs">Contact</Nav.Link>
      </Nav.Item>
      <Nav.Item>
      <Nav.Link href="/login">Login</Nav.Link>
      </Nav.Item>
    </Nav>
  </Navbar.Collapse>
  <Form>
  <Form.Control  as="select">
  <option>English</option>
  <option>Francais</option>
  <option>Kinyarwanda</option>
  </Form.Control>
  </Form>
  </Navbar>
  );
  }
};
