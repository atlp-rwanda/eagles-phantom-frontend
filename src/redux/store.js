import {createStore, applyMiddleware } from 'redux';
import { composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import allReducers from './reducers/';

// const rootReducer = combineReducers({
//   count: counterReducer,
// });

// const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const middleware = applyMiddleware(thunk, promise)
const store = createStore(allReducers, {}, composeWithDevTools(middleware));

export default store;
