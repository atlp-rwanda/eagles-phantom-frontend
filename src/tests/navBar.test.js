import React from 'react';
import { Provider } from "react-redux";
import store from "../redux/store.js";
import { render } from '@testing-library/react';
import Navbar from '../components/landingPage/Navbar';

const navBar=()=>{
    return render(
       <Provider store={store}>
       <Navbar />
       </Provider>
    )
}

describe("Navbar",()=>{
    it("should render the component",()=>{
        const { asFragment }=navBar();
        expect(asFragment(<Navbar />)).toMatchSnapshot()
    });
})