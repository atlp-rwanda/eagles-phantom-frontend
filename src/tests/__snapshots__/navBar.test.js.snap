// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
<DocumentFragment>
  <nav
    class="nav navbar navbar-expand-lg navbar-dark"
  >
    <a
      class="navbar-brand"
      href="#home"
    >
      Phantom
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="justify-content-end navbar-nav"
      >
        <div
          class="nav_link nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="#"
            href="#"
            role="button"
          >
            Home
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="AboutUs"
            href="AboutUs"
          >
            About
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="/ContactUs"
            href="/ContactUs"
          >
            Contact
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="/login"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
    <form
      class=""
    >
      <select
        class="form-control"
      >
        <option>
          English
        </option>
        <option>
          Francais
        </option>
        <option>
          Kinyarwanda
        </option>
      </select>
    </form>
  </nav>
</DocumentFragment>
`;
